public/app.js

```diff
@@ -0,0 +1,359 @@
+const APIController = (function() {
+
+  const clientId = 'e91988fcc1674aa88374f24be4844f85';
+  const clientSecret = '10badbb697f544ecb15e1efd7659a0b0';
+
+  const send = async (url,params) => {
+      let token = localStorage.my_token;
+
+      if (!token || (Date.now() - localStorage.token_at < localStorage.token_expiresIn)){
```

Comment:
Можно сделать проще
Можно в хранилице хранить время, когда истечет срок действия токена и сделать проверку типа
```suggestion
      if (!token || (Date.now() > localStorage.token_expiresIn)){
```